### PR TITLE
[Rec-IM] Team participant upgrades and fixes

### PR DESCRIPTION
**Pull request by amos-cha**
Tackles the following:

Team Page
--
- Participants are **deleted** if activity is still in registration period 
- Participants are made **inactive** if activity is out of registration period
^ this is done so that participants cannot jump teams as well as not locking themselves into a team during the registration period.

- Clicking off menu will not refresh the page
- Clicking any of the updates will refresh via callback

image
Member reinstation after made inactive

image
Old methodology now ACTUALLY works

image
Invited participants cant be removed or promoted

Bug Fixes
--
- Home page no longer pulls inactive participation as a part of home teams

### CODE REVIEW

File: src/services/recim/series.ts [hunk 1]
```diff
@@ -99,7 +99,7 @@ const putSeriesSchedule = async (schedule: UploadSeriesSchedule): Promise<Series
   http.put(`recim/series/schedule`, schedule);
 
 const scheduleSeriesMatches = async (seriesID: number): Promise<Match[]> =>
-  http.post(`recim/series/${seriesID}/autoschedule`);
+  http.post(`recim/series/${seriesID}/autoschedule`, {});
```
Comment (silasw2) [inline src/services/recim/series.ts:102]: Do we need a body tag for this?

File: src/views/RecIM/components/List/Listing/index.jsx [hunk 2]
```diff
@@ -365,15 +369,20 @@ const ParticipantListing = ({
...
   const moreOptionsOpen = Boolean(anchorEl);
   const [didAttend, setDidAttend] = useState(initialAttendance != null);
 
+  const handleClickOff = () => {
+    setAnchorEl(null);
+  };
+
   const handleClose = () => {
     setAnchorEl(null);
+    callbackFunction((val) => !val);
```
Comment (silasw2) [inline src/views/RecIM/components/List/Listing/index.jsx:385]: nice implementation of this!

File: src/views/RecIM/components/List/Listing/index.jsx [hunk 3]
```diff
@@ -409,12 +424,29 @@ const ParticipantListing = ({
...
+    await editTeamParticipant(parseInt(teamIDParam), editedParticipant);
+    handleClose();
+  };
+
+  const handleMakeInactive = async () => {
     let editedParticipant = {
       Username: participant.Username,
       RoleTypeID: 6,
     }; // Role 6 is inactive
+    await editTeamParticipant(parseInt(teamIDParam), editedParticipant);
```
Comment (silasw2) [inline src/views/RecIM/components/List/Listing/index.jsx:441]: was the parse int necessary here? would it work without?

**Review by silasw2 — APPROVED** [review]
LGTM! Ask me to re-review when you have the other PR ready!

File: src/services/recim/series.ts (hunk 1 above)
Comment (amos-cha) [reply src/services/recim/series.ts:102]: yep, the other PR adds a comment in there stating `temporary solution`. There is a new parameter to handle Round Robin Match Cap & Number of Ladder Matches

File: src/views/RecIM/components/List/Listing/index.jsx (hunk 3 above)
Comment (amos-cha) [reply src/views/RecIM/components/List/Listing/index.jsx:441]: I'm not sure if it's necessary as I haven't tried without it. It's based on the boiler plate set up with the old promote logic from when the horizon menu was first made.

**Comment by amos-cha** [comment]
Add minor new implementation:
image
Co-captains now can be demoted to member!

**Review by silasw2 — APPROVED** [review]
LGTM!

**Pull request merged**